### ISSUE TITLE
Is this workshop too big?

### ISSUE THREAD

**Issue by BartBucknill** [foundersandcoders/express-handlebars-workshop#16]
Both FAC11 and FAC12 (London) have found the size of this refactoring exercise confusing. The detailed instructions provided for FAC12 provided a better experience than that of FAC11, but there remained considerable confusion.

I think that the cognitive load resulting from the large number of files and large amount of content makes it difficult for students to focus on the core purpose of the workshop: learning handlebars. 

For future iterations perhaps we should consider, either:
* significantly reducing the size of the refactoring exercise

or:
* not having it as a refactoring exercise at all, but rather as a new build exercise.

**Comment by eliascodes** [comment]
@BartBucknill if we halve the number of fruits to 7, do think that is enough?

**Comment by BartBucknill** [comment]
@eliascodes I don't think that halving the number of fruits alone would have much effect, when you can do one you can do all of them. TBH, I'm not sure how best to reduce the size of the refactoring to reduce the cognitive load. Maybe it shouldn't be a refactoring exercise at all.

**Comment by jamiecoe** [comment]
@eliascodes I believe our updated instructions for #12 have made this issue irrelevant. Can it be closed?

**Comment by thejoefriel** [comment]
Hi @BartBucknill - on fac14 and doing curriculum review. I can see that this issue may have been resolved? Or you ok if this is closed?

**Comment by BartBucknill** [comment]
@thejoefriel Based on a quick look at the README it seems to me that this is still a large refactoring task. However, if you and your cohort believe that this issue is no longer relevant/the problem has been solved, please go ahead and close it 👍

**Comment by m4v15** [comment]
@arrested-developer @VirtualDOMinic @Armand-Lluka

Not sure if you guys agree, but I think @BartBucknill makes some good points re: this workshop - I'm not sure it does a _great_ job of introducing _why_ handlebars is a good tool to use, or even _how_ to use it well. I might not do this workshop in FACK1 and will try to develop another workshop introducing hbs, along these lines:

---

# Handlebars

## What is templating/SSR presentation

- Talk through flow of week 6 project
  - Request for home route (HTML), request for JS, request for data, manipulate the DOM
- Wouldn't it be cool if we did this alllll in one request and didn't have to manipulate the DOM
    - get the data on the home route straight away, insert it into the HTML
- Could do this easily enough - make a function that takes data, produces an HTML string out of it, send that

**Make presentation with live examples of the above**

## Handlebars!

Handlebars is just a simple way of doing exactly that without having to make a lot of functions

- Short express handlebars walkthrough - basic set up, render a couple of pages, pass in some context.

## Challenge
- Give them the current challenge/do the below as a challenge:
    - start with small app vanilla node app that is making requests and doing DOM manipulation
    - Convert into express handlebars app

---

I will share things with you guys as I make them, should be done by Saturday night at the latest.

**Comment by m4v15** [comment]
@arrested-developer @VirtualDOMinic @Armand-Lluka

This is my first swing at an introductory presentation if you want to use in london (see the presentation notes for more detail): 

https://docs.google.com/presentation/d/1AHAMQxpudl0zwwR2yADIoB7Luf58urJpKaqqTb1dhu0/edit?usp=sharing

**Comment by m4v15** [comment]
Addtionally, I've made this repo to compare various way's of doing a CMS:

https://github.com/m4v15/templating

**Comment by m4v15** [comment]
The presentation/example repo/workshop are now all finished if people are interested in checking them out (no worries if not)

**Comment by m4v15** [comment]
@Armand-Lluka @VirtualDOMinic 

Just did them here, they went really well, but they are definitely a bit long, here were my timings:

- Presentation, including walkthrough of templating repo **45 minutes**
- short break **5 minutes**
- Walkthrough for setting up handlebars ( I also got them started on challenge 1 - got them to the point of rendering a new view, but had them investigate how to use #each and #if **30 minutes**
- Up to the end of part 3 (some people started part 4) **50 minutes**

And then we had to stop. Notes:
- generally went well, a lot of chances to ask them questions and check understanding
- go very slow for fiddly setting up of express-handlebars
- people didn't know where to start for challenge 1, hence the walkthrough just made a new view and route, left them to research the helpers
- challenge 2 is just copy and pasting challenge 1 really, not a huge amount of point
- challenge 3, no one could find the bit in the docs which says how to set the `partialsDir` might just be better to show them - we were running out of time so I ended up just doing challenge 3 as a walkthrough as no one could get past setting up partials directory. Probably would be better to walk through making the partial and adding the directory in `exphbs()` config, and then have the students research it.

Feel free to ask anything here if you want more feedback. There is a danger that stuff here overlaps with stuff in the MC but I think just reinforcing things is not the worst thing in the world.

**Comment by VirtualDOMinic** [comment]
@m4v15 great job on the workshop! Largely went well with us, too. Some small changes/issues, some which overlap with yours (e.g. partialsDir). I'll make a PR for your repo at some point soon :)

**Comment by oliverjam** [comment]
@VirtualDOMinic when you say "went well with us" do you mean you didn't do the workshop in _this_ repo?

@m4v15 your workshop looks great. Are you still using it in place of this one? I'm not convinced this is a great intro as it's really complicated considering it's _just_ supposed to introduce Handlebars.

I've always thought we should start introducing this concept earlier on with template literals, since they're literally templating 🙄 

e.g. every project has some
```js
response.end("<h1>Server Error</h1>");
```
in it. Upgrade that to
```js
response.end(`<h1>Server error: ${error.message}</h1>`);
```
and boom, you're templating.

**Comment by m4v15** [comment]
@oliverjam I'm pretty sure @VirtualDOMinic used my workshops, but not 100%.

WRT is it still being used, I don't know - We used it for Khaleel, but I don't think it was used in the most recent Gaza round. I think FAC16 haven't got here yet (cc: @charlielafosse ) so don't know if they are planning on using it, but I think they probably are.

I'd say this is probably good to get merged, but then I am a bit biased.

I agree about about introducing templating earlier, but I guess that maybe requires a little bit more thought to find the space when (@mattlub suggests week 5 above) and how that would change the structure of the 8 weeks

**Comment by oliverjam** [comment]
FAC16 definitely did the workshop in this repo (there's an issue raised by one of them 2 days ago).

I think they struggled with it a bit, so maybe we should make the decision to switch over to your one officially until we can sort something out long-term?

**Comment by m4v15** [comment]
I agree - this is the schedule we used in Khalil (this is workshop 3)

https://github.com/m4v15/master-reference/tree/master/coursebook/week-8

**Comment by Armand-Lluka** [comment]
@oliverjam  @m4v15 

We definitely didn't use the fruits workshop, the broad consensus was that it felt too bloated trying to do too much. 

The feedback at the time from FAC15 on the alternative provided seemed mostly positive. I also recommend it gets merged.
